includes/widgets/playing-cards.php

```diff
@@ -0,0 +1,500 @@
+				'selectors' => [
+					'{{WRAPPER}} {{CURRENT_ITEM}}.e-playing-card' => '--e-card-border-color: {{VALUE}};',
+				],
+				'options' => [
+					'default' => esc_html__( 'Default Color', 'elementor' ),
+					'custom' => esc_html__( 'Custom', 'elementor' ),
+				],
+			],
+		);
+
+		$this->add_control(
+			'card_border_radius',
+			[
+				'label' => esc_html__( 'Cards Border Radius', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', 'rem', 'vw', 'custom' ],
+				'range' => [
+					'px' => [
+						'min' => 0,
+						'max' => 20,
+					],
+				],
+				'selectors' => [
+					'{{WRAPPER}} {{CURRENT_ITEM}}.e-playing-card' => '--e-card-border-radius: {{SIZE}}{{UNIT}}',
+				],
+			],
+		);
+
+		$this->add_control(
+			'card_number_color',
+			[
+				'label' => __( 'Cards Number Color', 'elementor' ),
+				'type' => Controls_Manager::COLOR,
+				'selectors' => [
+					'{{WRAPPER}} {{CURRENT_ITEM}}.e-card-number' => '--e-card-number-color: {{VALUE}};',
+				],
+				'options' => [
+					'default' => esc_html__( 'Default Color', 'elementor' ),
+					'custom' => esc_html__( 'Custom', 'elementor' ),
+				],
+			]
+		);
+
+		$this->add_control(
+			'card_suite_color',
+			[
+				'label' => __( 'Cards Suite Color', 'elementor' ),
+				'type' => Controls_Manager::COLOR,
+				'selectors' => [
+					'{{WRAPPER}} {{CURRENT_ITEM}}.e-card-suite' => '--e-card-suite-color: {{VALUE}};',
+				],
+				'options' => [
+					'default' => esc_html__( 'Default Color', 'elementor' ),
+					'custom' => esc_html__( 'Custom', 'elementor' ),
+				],
+			]
+		);
+
+		$this->add_responsive_control(
+			'card_size',
+			[
+				'label' => esc_html__( 'Cards Suite Size', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', 'rem', 'vw', 'custom' ],
+				'range' => [
+					'px' => [
+						'min' => 20,
+						'max' => 80,
+					],
+				],
+				'selectors' => [
+					'{{WRAPPER}} {{CURRENT_ITEM}}.e-card-suite' => '--e-card-suite-size: {{SIZE}}{{UNIT}}',
+				],
+			]
+		);
+
+		$this->add_control(
+			'card_padding',
+			[
+				'label' => esc_html__( 'Cards Padding', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', 'rem', 'vw', 'custom' ],
+				'range' => [
+					'px' => [
+						'min' => 0,
+						'max' => 20,
+					],
+				],
+				'selectors' => [
+					'{{WRAPPER}} {{CURRENT_ITEM}}.e-playing-card' => '--e-card-padding: {{SIZE}}{{UNIT}}',
+				],
+			],
+		);
+
+		$this->end_controls_section();
+	}
+
+
+	/**
+	 * Render Playing cards widget output on the frontend.
+	 *
+	 * Refactored to use helper methods for rendering controls and individual cards.
+	 *
+	 * @since 1.0.0
+	 * @access protected
+	 */
+	protected function render(): void {
+		$settings = $this->get_settings_for_display();
+
+		if ( empty( $settings['cards'] ) ) {
+			return;
+		}
+
+		$this->render_card_controls();
+
+		echo '<div class="e-playing-cards-container e-flex card-section">';
```

Comment:
We can open a close ` <?php ?>` to avoid an echo.

```suggestion
		<div class="e-playing-cards-container e-flex card-section">
```

Reply:
fixed